Index: grails-app/views/_tabs.gsp
===================================================================
--- grails-app/views/_tabs.gsp	(revision 29e6950d6f01e6f7523883a1b06d519b52edc6fe)
+++ grails-app/views/_tabs.gsp	(revision )
@@ -1,9 +1,8 @@
 <%@ page contentType="text/html;charset=UTF-8" %>
 <ul id="main-tabs">
-	<li><g:link
-			class="tab ${!(params['archived']) && ['message','folder','poll'].contains(params.controller)?'selected':''}" url="${[controller:'message']}"
+	<li><g:link class="tab ${(!params['archived'] || params['archived'] == 'false') && ['message','folder','poll'].contains(params.controller)?'selected':''}" url="${[controller:'message']}"
 			id="tab-messages">Messages (${frontlinesms2.Fmessage.countUnreadMessages()})</g:link></li>
-	<li><g:link class="tab ${params['archived'] ? 'selected':''}" controller='message' action="inbox" params="['archived': true]">Archive</g:link></li>
+	<li><g:link class="tab ${params['archived'] == 'true'? 'selected':''}" controller='message' action="inbox" params="['archived': true]">Archive</g:link></li>
 	<li><g:link class="tab ${params.controller=='contact'?'selected':''}" url="${[controller:'contact']}" id="tab-contacts">Contacts</g:link></li>
 	<li><g:link class="tab ${params.controller=='status'?'selected':''}" url="${[controller:'status']}" id="tab-status"><img id='indicator'/> Status</g:link></li>
 	<li><g:link class="tab ${params.controller=='search'?'selected':''}" url="${[controller:'search']}" id="tab-search">Search</g:link></li>
Index: grails-app/views/layouts/archive.gsp
===================================================================
--- grails-app/views/layouts/archive.gsp	(revision 29e6950d6f01e6f7523883a1b06d519b52edc6fe)
+++ grails-app/views/layouts/archive.gsp	(revision )
@@ -7,12 +7,16 @@
 	<link rel="shortcut icon" href="${resource(dir: 'images', file: 'favicon.ico')}" type="image/x-icon"/>
 	<g:javascript library="jquery" plugin="jquery"/>
 	<jqui:resources/>
+	<g:javascript src="message/actions.js"></g:javascript>
+	<g:javascript src="application.js"/>
+	<g:javascript src="popup.js" />
+	<g:javascript src="/message/move_dropdown.js" />
 </head>
 <body>
 <g:render template="/tabs"/>
 <g:render template="/flash"/>
 <div id="main">
-	<div id="content">
+	<div id="archive">
 		<g:render template="../archive/menu"/>
 		<g:render template="message_list"/>
 		<g:layoutBody/>
Index: grails-app/views/message/_message_details.gsp
===================================================================
--- grails-app/views/message/_message_details.gsp	(revision 29e6950d6f01e6f7523883a1b06d519b52edc6fe)
+++ grails-app/views/message/_message_details.gsp	(revision )
@@ -1,14 +1,15 @@
-<g:hiddenField id="message-src" name="message-src" value="${messageInstance.src}" />
+<g:hiddenField id="message-src" name="message-src" value="${messageInstance.src}"/>
-<g:hiddenField id="message-id" name="message-id" value="${messageInstance.id}" />
+<g:hiddenField id="message-id" name="message-id" value="${messageInstance.id}"/>
 <div id="message-details">
 	<p class="message-name">${messageInstance.displayName}</p>
-	<g:def var="thisAddress" value="${messageInstance.src}" />
+	<g:def var="thisAddress" value="${messageInstance.src}"/>
 	<g:if test="${!messageInstance.contactExists}">
 		<g:link class="button" controller="contact" action="createContact" params="[primaryMobile: thisAddress]">+</g:link>
 	</g:if>
-	<p class="message-date"><g:formatDate format="dd-MMM-yyyy hh:mm" date="${messageInstance.dateCreated}" /></p>
+	<p class="message-date"><g:formatDate format="dd-MMM-yyyy hh:mm" date="${messageInstance.dateCreated}"/></p>
 	<p class="message-body" id="message-body">${messageInstance.text}</p>
-	<div class="buttons">
+
+	<div class="single-action buttons">
 		<g:if test="${messageSection != 'trash'}">
 			<button id="btn_reply">Reply</button>
 			<button id='btn_dropdown'></button>
@@ -20,4 +21,20 @@
 		</g:if>
 	</div>
 </div>
+
+<div class="multi-action buttons hide">
+	<div id="count"></div>
+	<g:if test="${messageSection != 'pending'}">
+		<a id='btn_reply_all'>Reply All</a>
+	</g:if>
+	<g:if test="${!params['archived']}">
+		<a id='btn_archive_all'>Archive All</a>
+	</g:if>
+	<a id='btn_delete_all'>Delete All</a>
+</div>
+
+
+<g:if test="${!params['archived']}">
-<g:render template="/message/action_list"/>
+	<g:render template="/message/action_list"/>
+</g:if>
+
Index: web-app/js/message/actions.js
===================================================================
--- web-app/js/message/actions.js	(revision 29e6950d6f01e6f7523883a1b06d519b52edc6fe)
+++ web-app/js/message/actions.js	(revision )
@@ -42,14 +42,12 @@
 	highlightRow(id);
 
 	var count = countCheckedMessages();
-	if(count == 1 && $(':checkbox').size() != 2) {
-		if(selectedMessageId != $('tr :checkbox[checked="true"]').val())
+	if(count == 1) {
-			loadMessage($('tr :checkbox[checked="true"]').val(), true);
+		loadMessage($('tr :checkbox[checked="true"]').val(), true);
 	}
-	if(count > 1){
+	if(count > 1)
 		changeMessageCount(count);
-	}
+}
-}
 
 function loadMessage(id, checked) {
 	var location = window.location.pathname
@@ -86,31 +84,20 @@
 
 function changeMessageCount(count){
 	setSelectedMessage();
-	$('#message-details').empty();
-	$('#message-details').append("<p> "+count+" messages selected</p>");
+	$('#count').html("<p> "+count+" messages selected</p>");
 	setMessageActions();
 }
 
 function setMessageActions() {
-	var replyAll = '';
-	var messageSection = $('input:hidden[name=messageSection]').val()
-	if(messageSection != 'pending'){
-		replyAll = "<a id='btn_reply_all' >Reply All</a>";
+	$('.multi-action').show()
+	$('#message-details').hide()
-	}
+}
-	archiveAll = "<a id='btn_archive_all' >Archive All</a>";
-	deleteAll = "<a id='btn_delete_all' >Delete All</a>";
-	$('#message-details').append("<div class='buttons'></div>");
-	$('#message-details div.buttons').append(replyAll+"&nbsp;"+archiveAll+"&nbsp;"+deleteAll);
-	$('#btn_reply_all').click(quickReplyClickAction);
-	$('#btn_delete_all').click(deleteAllClickAction);
-	$('#btn_archive_all').click(archiveClickAction);
-}
 
 function getSelectedGroupElements(groupName) {
 	return $('input[name=' + groupName + ']:checked');
 }
 
-function quickReplyClickAction() {
+$('#btn_reply_all').live('click', function() {
 	var me = $(this);
 	var messageType = me.text();
 
@@ -127,9 +114,9 @@
 		url: '/frontlinesms2/quickMessage/create',
 		success: function(data, textStatus){ launchWizard(messageType, data); }
 	});
-}
+});
 
-function deleteAllClickAction() {
+$('#btn_delete_all').live('click', function() {
 	var me = $(this);
 	var messageType = me.text();
 	var idsToDelete = []
@@ -148,9 +135,9 @@
 		url: '/frontlinesms2/message/deleteMessage',
 		success: function(data) { reloadPage(messageSection, ownerId)}
 	});
-}
+});
 
-function archiveClickAction() {
+$('#btn_archive_all').live('click', function() {
 	var me = $(this);
 	var messageType = me.text();
 	var messageSection = $('input:hidden[name=messageSection]').val();
@@ -169,7 +156,7 @@
 		url: '/frontlinesms2/message/archiveMessage',
 		success: function(data, textStatus){ reloadPage(messageSection, ownerId)}
 	});
-}
+});
 
 function reloadPage(messageSection, ownerId) {
 	if(messageSection == 'poll' || messageSection == 'folder'){
@@ -185,12 +172,17 @@
 	}
 }
 
+function enableSingleAction() {
+	$('.multi-action').hide()
+	$('#message-details').show()
+}
 function showMessageDetails(){
-	if(selectedMessageId == null){
-		return;
+//	if(selectedMessageId == null){
+//		return;
+//	}
+//	loadMessage(selectedMessageId,false);
+	enableSingleAction();
-	}
+}
-	loadMessage(selectedMessageId,false);
-}
 
 function highlightRow(id){
 	if( $('tr :checkbox[value='+id+']').attr('checked') == 'checked'){
